Library/Homebrew/formula.rb

```diff
@@ -1013,9 +1018,13 @@ def _outdated_versions(options = {})
 
     installed_kegs.each do |keg|
       version = keg.version
+      tab = Tab.for_keg(keg)
       all_versions << version
 
-      return [] if pkg_version <= version && !version.head?
+      latest_version = pkg_version <= version
+      same_version_scheme = version_scheme == tab.version_scheme
+      not_head = !version.head?
+      return [] if latest_version && same_version_scheme && not_head
```

Comment:
Actually, I just realize I give an incorrect suggestion in the first place. The correct way is:
```ruby
installed_kegs.each do |keg|
  version = keg.version
  tab = Tab.for_keg(keg)
  all_versions << version
  next if version.head?
  next if version_scheme > tab.version_scheme
  next if version_scheme == tab.version_scheme && pkg_version > version
  return []
end
```

Noted that the correct way to compare two version is:
```ruby
(version_scheme <=> other.version_scheme).nonzero? || pkg_version <=> other.pkg_version
```